### PR TITLE
python: Collect and report error messages from sudo.

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
This is a UI improvement and should also result fewer error messages landing in the journal.

### CODE REVIEW

File: src/cockpit/superuser.py [hunk 1]
```diff
@@ -135,7 +137,17 @@
...
-                self.startup.failed(self, exc or ConnectionResetError('connection lost'))
+                if exc is None:
+                    # The connection closed down without a specific failure.
+                    # If we have stderr from a spawned program, then use it as
+                    # an error message.  Otherwise, return a generic error.
+
+                    if isinstance(peer.transport, SubprocessTransport):
+                        error_message = peer.transport.get_stderr()
+
+                    exc = RuntimeError(error_message or 'connection lost')
```
[the remark this answers is not in the document]
Comment (allisonkarlitskaya) [reply src/cockpit/superuser.py:150]: Thanks, bot.

**Review by jelly — APPROVED** [review]

**Comment by allisonkarlitskaya** [comment]
This needs a lot more work before the tests start to pass.

 - we need to also collect the "lecture" messages and pass them as part of the prompts
 - we need to do the trickery with forwarding stderr to the spawned bridge, if it requests it

**Comment by jelly** [comment]
Do we want to merge it with follow ups? Seems the tests are happy at least? Or shall we trigger a run against the `TEST_DEFAULT_OS` bump first if it actually helps?

**Comment by allisonkarlitskaya** [comment]
This has gotten complex enough that @mvollmer should definitely take a look at it.

**Review by mvollmer — APPROVED** [review]
This is simpler than I feared. Thanks!

**Comment by martinpitt** [comment]
Yay, seems to [fix one test](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18132-20230106-130050-dbc00eb4-fedora-36-pybridge/log.html#246). This should also allow us to drop some expected messages in testlib.py?

**Comment by allisonkarlitskaya** [comment]
> Yay, seems to [fix one test](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18132-20230106-130050-dbc00eb4-fedora-36-pybridge/log.html#246). This should also allow us to drop some expected messages in testlib.py?


Oops.  For some reason I thought that this list was kept in the bots.  Just added a commit to remove the sudo-related messages.  Let's see how it does.

File: src/cockpit/peer.py [hunk 2]
```diff
@@ -96,15 +96,31 @@ def do_closed(self, transport_was: asyncio.Transport, exc: Optional[Exception])
             self.send_channel_control(self.channels.pop(), 'close', problem='disconnected')
 
         if self.state_listener is not None:
+            # If we don't otherwise has an exception set, but we have stderr output, we can use it.
```
Comment (mvollmer) [inline src/cockpit/peer.py:103]: s/has/have/, if you push again.

**Review by mvollmer — APPROVED** [review]

**Pull request merged**